checker packaging

skas102 opened the pull request.

Commit:
checker packaging

tk-codes commented on checker/checker.go:
```diff
@@ -27,15 +30,15 @@ func New(syntaxTree *node.ProgramNode) *Checker {
 // Returns the symbol table and errors
 func (c *Checker) Run() (*symbol.SymbolTable, []error) {
-	c.symbolTable, c.errors = phase.RunSymbolConstruction(c.syntaxTree)
+	c.symbolTable, c.errors = symbolconstruction.RunSymbolConstruction(c.syntaxTree)
```
You can now simplify the function `symbolconstruction.Run(...)`, since the phases are in different package.

Do it for the rest of the phases as well.

tk-codes commented on checker/designatorresolution/designator_resolution.go:
```diff
@@ -1,8 +1,7 @@
-package phase
+package designatorresolution
```
Add a simple package doc. See lexer, parser description in https://godoc.org/github.com/bazo-blockchain/lazo

Do it for all packages

tk-codes commented on checker/designatorresolution/designator_resolution.go:
```diff
@@ -23,7 +22,7 @@ func RunDesignatorResolution(symTable *symbol.SymbolTable) []error {
 func (dr *designatorResolution) resolveDesignators() {
 	contractSymbol := dr.symTable.GlobalScope.Contract
-	v := visitor.NewDesignatorResolutionVisitor(dr.symTable, contractSymbol)
+	v := NewDesignatorResolutionVisitor(dr.symTable, contractSymbol)
```
`NewDesignatorResolutionVisitor` function can be private now, can't be?

Fix it for all New visitor methods.

tk-codes commented on checker/typecheck/type_check_visitor.go:
```diff
@@ -1,4 +1,4 @@
-package visitor
+package typecheck
```
Make `TypeCheckVisitor` private.

Same for all other checker visitors.

Commit:
Fix Concerns:

- Added package doc
- Made visitors private
- Made visitor creation private
- Renamed Run functions
